File: 1. Two Sum/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(nums)):
            complement = target-nums[i]
            if complement in map:
                return [i,map[complement]]
            map[nums[i]] = i
        return []
        

File: 1. Two Sum/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> um;
        vector<int> index;
        for(int i=0;i<nums.size();i++)
        {
            auto find = um.find(target-nums[i]);
            if(find != um.end() )
            {
                index.push_back(i);
                index.push_back(find->second);
                return index;
            }
            um[nums[i]] = i;
        }
        
        return index;
    }
};